### DISCUSSION TITLE
sd_skip_if also for questions

### DISCUSSION THREAD

**Discussion by StefanMunnes** [surveydown-dev/surveydown#124] [Feature Requests]
### Description

I want to hide a question if there are some previous specific answers, but show it if it has been answered with the other options or not answered at all. At the moment, I can solve it like this:
```r

sd_show_if(
  input$interessted_buying_car == "yes" || length(input$interessted_buying_car) == 0 ~ "car_preference"
)
```
But this is very inconvenient and error-prone. Doesn't it make sense to put the equivalent into the `sd_skip_if()` function? I thought that might be the simplest solution in terms of logic.

**Comment by jhelvy** [comment]
First, do you mean `sd_show_if`? You have `sd_skip_if` in your question but are description `sd_show_if`, so I'm going to guess that was just a typo. `sd_skip_if` sets the conditions to skip to target pages, so when you click a next button it will skip to a target page if the condition is met.

What I think you're asking about is the ability to check if a question is answered or not, which would make your logic a little easier. I just opened #122 to address this, so if we add that your example would look like this:

```r
sd_show_if(
  input$interessted_buying_car == "yes" || !sd_is_answered(input$interessted_buying_car) ~ "car_preference"
)
```

It's not necessarily shorter, but it would be more robust as the `sd_is_answered()` function can be made more robust to apply to different types of questions.

**Reply by StefanMunnes** [reply]
No, `sd_skip_if()` is correct. I know, for now it's to skip pages and a possible `sd_is_answered()` function will make things more easy, but wouldn't it be possible and useful to "open" the `sd_skip_if()` function also for questions? This would lead to more straight forward logic:
```r
sd_skip_if(
  input$interessted_buying_car == "no" ~ "car_preference" # will skip question with ID "car_preference"
)
```

**Reply by jhelvy** [reply]
Oh I see, yeah I misunderstood your question. So you're asking if the `target` in the `sd_skip_if` condition could support both page and question ids. That should be do-able. We keep track of which questions are on which page, so all we'd have to do is first check if the provided `target` is a page_id (and go to the page if it is); if not, then check if it's a `question_id` then go to the page that question is on. We could probably also move the focus to that question automatically too. Does that sound like what you would want?

The only condition I think this would require is that you do not use the same `id` for questions and pages, but that's easy enough to avoid.
